http-api/src/main/java/bisq/http_api/auth/HttpApiAuthFilter.java

```diff
@@ -0,0 +1,47 @@
+package bisq.http_api.auth;
+
+import bisq.common.encoding.Hex;
+import bisq.http_api.config.CommonApiConfig;
+import bisq.security.DigestUtil;
+import jakarta.annotation.Priority;
+import jakarta.ws.rs.Priorities;
+import jakarta.ws.rs.container.ContainerRequestContext;
+import jakarta.ws.rs.container.ContainerRequestFilter;
+import jakarta.ws.rs.core.Response;
+import jakarta.ws.rs.ext.Provider;
+import lombok.extern.slf4j.Slf4j;
+
+import java.io.IOException;
+import java.security.MessageDigest;
+
+@Provider
+@Priority(Priorities.AUTHENTICATION)
+@Slf4j
+public class HttpApiAuthFilter implements ContainerRequestFilter {
```

Comment:
Nit: Maybe use same name pattern for HttpApiAuthFilter and WebSocketAuthenticationFilter